ci/config/configuration.py

```diff
@@ -0,0 +1,228 @@
+                        valid = False
+                    else:
+                        print(f"{cstr} {util.colorize('PASS', 'green')}")
+        print()
+
+        # Verify clusters
+        print(util.colorize("Validating cluster configurations", "blue"))
+        for name, cluster in self._clusters.items():
+            cstr = f"{name}"
+            # check that there is one partition for every uarch
+            if len(cluster["partition"]) != len(cluster["uarch"]):
+                print(f"{cstr:25s} {util.colorize('FAIL', 'red')} there must be exactly one partition for each uarch")
+                valid = False
+            # check that the FirecREST runner hasn't been selected (not supported yet)
+            elif cluster["runner"] == "f7t":
+                print(f"{cstr:25s} {util.colorize('WARN', 'cyan')} the FirecREST 'f7t' runner is not supported yet")
+            else:
+                print(f"{cstr:25s} {util.colorize('PASS', 'green')}")
+        print()
+
+        if not valid:
+            raise ConfigError("configuration error - see log")
+
+    def is_valid_target(self, cluster, uarch):
+        if cluster not in self._clusters.keys():
+            return False, f"cluster {cluster} is not defined"
+        if uarch not in self._clusters[cluster]["uarch"]:
+            return False, f"cluster {cluster} does not support {uarch}"
+        return True, ""
+
+    def uenv(self, name):
+        """
+        return the uenv information for uenv with name.
+        returns None if no uenv matches name
+        """
+        for u in self.uenvs:
+            if u.name == name:
+                return u
+        return None
+
+    def recipe(self, name, version, uarch):
+        """
+        return the recipe information for uenv name:version on target uarch.
+        returns None if no recipe fits the description.
+        """
+        u = self.uenv(name)
+        if u is not None:
+            for v in u.versions:
+                if v.name==version and uarch in v.uarch:
+                    return v.recipe(uarch)
+
+        return None
+
+    @property
+    def uenvs(self):
+        return self._uenvs
+
+    @property
+    def clusters(self):
+        return self._clusters
+
+    def job_template(self, env):
+        """
+        returns a dict that contains the information required to configure
+        a gitlab runner job to build a recipe.
+        The input is a dictionary with the following fields (with example values):
+            system: clariden
+            uarch: a100
+            uenv: gromacs
+            version: 2023
+            recipe: /home/bcumming/software/github/alps-spack-stacks/recipes/gromacs/2023/a100
+        """
+        c = self.clusters[env["system"]]
+        part_idx = c["uarch"].index(env["uarch"])
+
+        develop = ""
+        version = self.uenv(env["uenv"]).version(env["version"])
+        if version.spack_develop:
+            develop = "--develop"
```

Comment:
I think the the correct option for `stack-config` here is `-d`, while `--develop` is for the `stackinator`. This produces the following [error](https://gitlab.com/cscs-ci/ci-testing/webhook-ci/mirrors/551234120955960/1440398897047560/-/jobs/6726335473#L25):
```
./stack-build: illegal option -- -
```